etc/bash_completion.d/brew

```diff
@@ -45,23 +45,9 @@ __brewcomp ()
 __brew_complete_formulae ()
 {
     local cur="${COMP_WORDS[COMP_CWORD]}"
-    local lib=$(brew --repository)/Library
-    local taps=${lib}/Taps
-    local ff=$(\ls ${lib}/Formula 2>/dev/null | sed 's/\.rb//g')
-    local af=$(\ls ${lib}/Aliases 2>/dev/null)
-    local tf file
-
-    for file in ${taps}/*/*/*.rb ${taps}/*/*/Formula/*.rb ${taps}/*/*/HomebrewFormula/*.rb; do
-        [ -f "$file" ] || continue
-        file=${file/"Formula/"/}
-        file=${file/"HomebrewFormula/"/}
-        file=${file#${lib}/Taps/}
-        file=${file%.rb}
-        file=${file/homebrew-/}
-        tf="${tf} ${file}"
-    done
-
-    COMPREPLY=($(compgen -W "$ff $af $tf" -- "$cur"))
+    local formulas=$(brew search)
+    local shortnames=$(echo "$formulas" | grep / | cut -d / -f 3)
```

Comment:
In theory they keep the output of the command together as a single argument, in practice Bash does some special casing for `local` and probably a few others. Always quoting when things are supposed to remain a single unit is easier to read in my opinion, because you don't have to remember those special cases all the time. In most other contexts, the result would be different without quoting, e.g.:

```
printf "[%s]\n" x=$(ls)
printf "[%s]\n" x="$(ls)"
```